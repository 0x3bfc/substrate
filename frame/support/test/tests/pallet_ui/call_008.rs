#[frame_support::pallet(Example)]
mod pallet {
	use frame_support::pallet_prelude::{ModuleInterface, DispatchResultWithPostInfo};
	use frame_system::pallet_prelude::{BlockNumberFor, OriginFor};

	#[pallet::trait_]
	pub trait Trait: frame_system::Trait {
		type Bar;
	}

	#[pallet::module]
	pub struct Module<T>(core::marker::PhantomData<T>);

	#[pallet::module_interface]
	impl<T: Trait> ModuleInterface<BlockNumberFor<T>> for Module<T> {}

	#[pallet::call]
	impl<T: Trait> Module<T> {
		#[pallet::weight(0)]
		fn foo(origin: OriginFor<T>, bar: T::Bar) -> DispatchResultWithPostInfo {
			Ok(().into())
		}
	}
}

fn main() {
}
