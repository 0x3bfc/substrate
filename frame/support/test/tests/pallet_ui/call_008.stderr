error[E0277]: the trait bound `pallet::Call<T>: pallet::_::_parity_scale_codec::Decode` is not satisfied
  --> $DIR/call_008.rs:18:2
   |
18 |     impl<T: Trait> Call for Module<T> {
   |     ^^^^^^^^^^^^^^^^^^^ the trait `pallet::_::_parity_scale_codec::Decode` is not implemented for `pallet::Call<T>`

error[E0277]: the trait bound `pallet::Call<T>: pallet::_::_parity_scale_codec::Encode` is not satisfied
  --> $DIR/call_008.rs:18:2
   |
18 |     impl<T: Trait> Call for Module<T> {
   |     ^^^^^^^^^^^^^^^^^^^ the trait `pallet::_::_parity_scale_codec::Encode` is not implemented for `pallet::Call<T>`
